sympy/geometry/ellipse.py

```diff
@@ -1362,6 +1357,46 @@ def __new__(cls, *args, **kwargs):
 
         raise GeometryError("Circle.__new__ received unknown arguments")
 
+    def object_from_equation(self):
+        """Returns a Circle object from an equation of circle given as input.
+        If the equation of the circle is ax**2 + by**2 + gx + hy + c = 0, then
+        the input has to be of the following format: ax**2 + by**2 + gx + hy + c
+
+        Examples
+        ========
+
+        >>> from sympy import Circle
+        >>> from sympy.abc import x, y
+        >>> Circle.object_from_equation(x**2 + y**2 + 3*x + 4*y - 8)
+        Circle(Point2D(-3/2, -2), 3.77491721763537)
+
+        """
+
+        from sympy.abc import x, y
+        poly_x = poly(self, x)
+        poly_y = poly(self, y)
+        deg_x = degree(poly_x)
+        deg_y = degree(poly_y)
+        lead_x_coeff = LC(poly_x)
+        lead_y_coeff = LC(poly_y)
+
+        if deg_x == 2 and deg_y == 2:
+            if lead_x_coeff == lead_y_coeff:
+                center_x = solve(diff(self, x))[0]
+                center_y = solve(diff(self, y))[0]
+                center = Point(center_x, center_y)
+                r_square = expand(lead_x_coeff * (x - center_x) ** 2) + expand(
+                    lead_y_coeff * (y - center_y) ** 2) - self
+                radius = (r_square / lead_x_coeff) ** 0.5
+                if radius.is_real:
+                    return Circle(center, radius)
+                else:
+                    raise GeometryError("The given equation of circle has an imaginary radius")
```

Comment:
This exception could be raised by just checking the sign of `g**2 + f**2 - c**2` (corresponding to general form of equation of a circle) instead of calculating sqrt and then checking if it is imaginary.

Reply:
This exception could be raised by just checking the sign of `g**2 + f**2 - c` (corresponding to general form of equation of a circle) instead of calculating sqrt and then checking if it is imaginary.